src/components/generic/ApexLineChart.vue

```diff
@@ -0,0 +1,219 @@
+  props: {
+    categories: Array, // Should be of type string[]
+    series: Array, // Should be of type object[]
+    upper_range: Number, //upper bound of y axis
+    sev_chart: Boolean, //identifies chart as severity chart
+    title: String,
+    y_title: String
+  }
+});
+
+let id_counter = 0;
+function next_id(): number {
+  id_counter += 1;
+  return id_counter;
+}
+
+export interface SeriesItem {
+  name: string;
+  data: number[];
+}
+
+/**
+ * Emits "category-selected" with payload of type Category whenever a category is selected.
+ */
+@Component({
+  components: {
+    VueApexCharts
+  }
+})
+export default class ApexLineChart extends ApexLineChartProps {
+  chart_id: string = `line_chart_${next_id}`;
+
+  get _categories(): string[] {
+    // Ensure it's an array
+    if (!(this.categories instanceof Array)) {
+      throw new Error('series must be an array of strings');
+    }
+
+    // Ensure all of its elements are numbers
+    this.categories.forEach(element => {
+      if (typeof element !== 'string') {
+        throw new Error(`Invalid series item ${element}`);
+      }
+    });
+
+    // We now know the type is definitely number[]
+    let final = this.categories as string[];
+
+    // If we have any non-zero data, just returngive 0.01 of all
+    return final;
+  }
+
+  /**
+   * Provide a type-checked accessor to our series property
+   */
+  get _series(): SeriesItem[] {
+    // Ensure it's an array
+    if (!(this.series instanceof Array)) {
+      throw new Error('series must be an array of objects');
+    }
+
+    // Ensure all of its elements are numbers
+    for (let element of this.series) {
+      if (typeof element !== 'object') {
+        throw new Error(`Invalid series item ${element}`);
+      }
+    }
+
+    // We now know the type is definitely number[]
+    let final = this.series as SeriesItem[];
+
+    // If we have any non-zero data, just returngive 0.01 of all
+    return final;
+  }
+
+  get label_colors(): string[] {
+    let colors = [];
+    for (let i = 0; i < this._categories.length; i++) {
+      colors.push('#FFFFFF');
+    }
+    return colors;
+  }
+
+  //creates differing number of ticks based on number of controls
+  get y_axis_tick(): number {
+    if (this.upper_range < 15) {
+      return this.upper_range;
+    } else if (this.upper_range < 50) {
+      return Math.floor(this.upper_range / 5);
+    } else {
+      return Math.floor(this.upper_range / 10);
+    }
+  }
+
+  //gives apex charts the severity colors
+  get sev_colors(): string[] {
+    let colors = [];
+    colors.push('#FFEB3B');
+    colors.push('#FF9800');
+    colors.push('#FF5722');
+    colors.push('#F44336');
+    return colors;
+  }
+
+  get line_colors(): string[] | undefined {
+    let colors = [];
+    if (this.sev_chart) {
+      return this.sev_colors;
+    }
+    return undefined;
+  }
+
+  //changes text based on light or dark mode
+  get white_black(): string {
+    var whiteBlack;
```

Comment:
Is this variable used anywhere?

Reply:
yes, line 172, 197, 204, 209

Reply:
That uses the `white_black` method however I don't see anything getting or setting the `whiteBlack` variable.